File: src/lib/types/db/profil/Profil.ts
export type ProfilRole = 'user' | 'admin' | 'guest';

export interface Profil {
  id: number;
  name: string;
  avatar: string | null;
  color: string;
  bio: string | null;
  role: ProfilRole;
  is_active: boolean;
  created_at: string;
  updated_at: string | null;
}

File: src-tauri/src/entity/profil/profil.rs
use chrono::{DateTime, Utc};
use serde::{Deserialize, Serialize};
use sqlx::FromRow;

#[derive(Debug, Clone, Serialize, Deserialize, sqlx::Type)]
#[sqlx(type_name = "TEXT")]
#[sqlx(rename_all = "lowercase")]
pub enum ProfilRole {
    User,
    Admin,
    Guest,
}

#[derive(Debug, Clone, Serialize, Deserialize, FromRow)]
pub struct Profil {
    pub id: i64,
    pub name: String,
    pub avatar: Option<String>,
    pub color: String,
    pub bio: Option<String>,
    pub role: ProfilRole,
    pub is_active: bool,
    pub created_at: DateTime<Utc>,
    pub updated_at: Option<DateTime<Utc>>,
}